### PR TITLE
add timeout option, resolve #54

### LINKED ISSUE #54: Runtime error getting interpolated values for long period/high interval
* PIconnect version: 0.5.1
* Python version: 2.7.13 |Anaconda custom (32-bit)| (default, May 11 2017, 14:07:41) [MSC v.1500 32 bit (Intel)]
* Operating System: Windows 10 Enterprise x64

### Description

Tried to get interpolated values for a long period (years) with high interval (minutes). System gives a runetime error

### What I Did
```
start = "01-01-2014"
eind = "13-03-2018"
interval = '2m'

for item in lijst: 
    with PI.PIServer() as server:
        tags = server.search(item)
        for tag in tags:
            data = tag.interpolated_values(start, eind, interval)
            data.to_csv(tag.tag + ".csv")
```
```
PITimeoutException: [-10722] PINET: Timeout on PI RPC or System Call.
   bij OSIsoft.AF.PI.PIException.ConvertAndThrowException(PIServer piServer, Exception ex, String message)
   bij OSIsoft.AF.PI.PIPoint.InterpolatedValuesByCount(AFTimeRange timeRange, Int32 numberOfValues, String filterExpression, Boolean includeFilteredValues)
   bij OSIsoft.AF.PI.PIPoint.InterpolatedValues(AFTimeRange timeRange, AFTimeSpan interval, String filterExpression, Boolean includeFilteredValues)
```

### PR DESCRIPTION
**Pull request by AlcibiadesCleinias**
add timeout option as discussed in https://github.com/Hugovdberg/PIconnect/issues/54

### CODE REVIEW

**Comment by Hugovdberg** [comment]
Thank you for this pull request, it looks like a good addition that should have been added a long time ago. However, I think I'll first have to fix some issues with Travis for the automated testing before I can merge it, hopefully I can get to that over the weekend.

**Comment by AlcibiadesCleinias** [comment]
> Thank you for this pull request, it looks like a good addition that should have been added a long time ago. However, I think I'll first have to fix some issues with Travis for the automated testing before I can merge it, hopefully I can get to that over the weekend.

oh, np. 
Thanks for the reply. I Hope Travis will stop hindering and let us even to release and pckg update... with your help ofc (;

**Review by Hugovdberg — APPROVED** [review]

**Comment by Hugovdberg** [comment]
I'm going to close and reopen this PR to trigger the pyup.io security checks

File: PIconnect/PI.py [hunk 1]
```diff
@@ -110,6 +115,11 @@ def __init__(
             self._credentials = None
         self.connection = self.servers.get(server, self.default_server)
 
+        if timeout:
+            from System import TimeSpan
+
+            self.connection.ConnectionInfo.OperationTimeOut = TimeSpan(0, 0, timeout)  # hour, min, sec
```
Comment (Hugovdberg) [inline PIconnect/PI.py:121]: The black formatting tool thinks this line is too long, perhaps we can integrate the comment in the code by changing it to something like this:

```python
hours, minutes, seconds = 0, 0, timeout
self.connection.ConnectionInfo.OperationTimeOut = TimeSpan(hours, minutes, seconds)
```

Please run `black PIconnect` from the root of the project afterwards to make sure it is within the currently configured
line length.

**Review by Hugovdberg — CHANGES_REQUESTED** [review]
@AlcibiadesCleinias Ok, finally getting some feedback from the build system on this PR. Even though I did approve the changes it appears the `black` formatting tool wants to see some changes. I think by making my suggested change we make the code a little more explicit, reducing the need of a comment (even though the line length might still be slightly too long). If you run `black PIconnect` from the root of the project the rest of the build should succeed just fine after that. Thanks for the useful contribution!

File: PIconnect/PI.py (hunk 1 above)
Comment (AlcibiadesCleinias) [reply PIconnect/PI.py]: Hola, @Hugovdberg,
not sure that it is normal to "create" more args than we need, e.g.
```
def foo():
    a, b = 0, 0
    return a, b

def not_a_foo():
    return 0, 0
```

I believe such one-time used arguments do not require vars, - only comment. So I relocate the comment

**Comment by Hugovdberg** [comment]
I'm working on a fix to get the builds working for this PR as well, currently you can't access the Codacy token, but when #582 is complete it should work for PRs from a fork as well.

**Review by Hugovdberg — APPROVED** [review]

**Comment by Hugovdberg** [comment]
I disabled the pyup check for the moment, I have no idea why it didn't get triggered.

**Pull request merged**